client_apis/python/src/tests/integration/watchlist_test.py

```diff
@@ -0,0 +1,420 @@
+#
+# CARBON BLACK API TESTS - watchlist
+# Copyright, Bit9, Inc 2015
+#
+
+""" These tests require CarbonBlack Enterprise Server to be installed.
+    You can run this script by passing the server URL and user API token
+    as parameters.
+"""
+
+import unittest
+import sys
+import os
+import requests
+from datetime import datetime
+
+if __name__ == '__main__':
+    sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../")))
+
+from cbapi.cbapi import CbApi
+
+cb = None
+name_prefix = "Test Watchlist"
+
+
+class CbApiWatchlistTest(unittest.TestCase):
+    def test_get_watchlists(self):
+        watchlists = cb.watchlist()
+        self.assertIsNotNone(watchlists)
+        return
+
+    def test_add_watchlist(self):
+        old_watchlists = cb.watchlist()
+        self.assertIsNotNone(old_watchlists)
+        add_result = cb.watchlist_add(
+            type="modules",
+            name="%s %s" % (name_prefix, datetime.now().isoformat(' ')),
+            search_query="q=is_executable_image%3Afalse&cb.urlver=1&sort=server_added_timestamp%20desc",
+            readonly=False
+        )
+        self.assertGreater(add_result["id"], 0)
+
+        new_watchlists = cb.watchlist()
+        self.assertEqual(len(new_watchlists), len(old_watchlists) + 1)
```

Comment:
I will definitely add that check, but I am going to keep the length check because I'd hope tests are not running concurrently and I'd want to catch a "double submission" type error.